public/coffee/ide/editor/directives/aceEditor.coffee

```diff
@@ -374,6 +374,21 @@ define [
 					if scope.eventsBridge?
 						session.on "changeScrollTop", onScroll
 
+					$rootScope.hasLintingError = false
+					session.on('changeAnnotation', () ->
+						# Both linter errors and compile logs are set as error annotations,
+						# however when the user types something, the compile logs are
+						# replaced with linter errors. When we check for lint errors before
+						# autocompile we are guaranteed to get linter errors
+						hasErrors = session
+							.getAnnotations()
+							.filter((annotation) -> annotation.type == 'error')
```

Comment:
typing a command with an unclosed bracket like `\foo{...` comes up as a warning `unclosed group {` while the user is typing it.

I think it's better if we also don't run autocompile for `type == 'warning'` as well as `'error'`  as this will not give a good result when the group hasn't been closed.   

We can filter out `type == 'info'` which only happens for things like content after \end{document}

Reply:
Good point. Pushed a fix